server/src/main/java/pl/edu/agh/server/survey/SurveyService.java

```diff
@@ -17,11 +19,20 @@ public String createSurvey(List<Term> terms){
         return survey.linkCode;
     }
 
-    public Survey getSurvey(String surveylinkCode){
+    public List<TermDTO> getSurvey(String surveylinkCode){
         Survey survey = surveyRepository.findBylinkCode(surveylinkCode);
         if (survey == null) {
             throw new IllegalStateException("Survey with linkCode " + surveylinkCode + " not found");
         }
-        return survey;
+
+        List<Term> termList = survey.getTermlist();
+        List<TermDTO> termDTOList = new ArrayList<>();
+
+        for (Term term : termList) {
+            TermDTO termDTO = new TermDTO(term);
```

Comment:
W sumie ciekawy sposob na mapowanie, raczeeej nie robimy tak, ale nie umiem w jave dobrze, jak dla mnie jest git i bym zostawil bo dziala XD
@kBlazewicz

Reply:
te przypisania termList = i termDTO= są wsumie niepotrzebne bo wiadomo o co chodzi działa  działa więc czemu nie
można by od 27 do 36 liniujki zastąpić  czymś takim +- ale to tylko ciekawostka szkoda czasu zebys się wczytywała co co robi a ja mogłem napisać z błędem bez IDE
return survey.getTermnilst().map(term -> new TermDTO(term));